### PR TITLE
zsh 5.3.1-test-2 (devel)

### PR DESCRIPTION
**Pull request by zmwangx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

```
 _ _                        _     
(_) | _____   _____ _______| |__  
| | |/ _ \ \ / / _ \_  / __| '_ \ 
| | | (_) \ V /  __// /\__ \ | | |
|_|_|\___/ \_/ \___/___|___/_| |_|
                                  
```

### CODE REVIEW

File: Formula/zsh.rb [hunk 1]
```diff
@@ -29,11 +29,18 @@ class Zsh < Formula
     sha256 "130c8b5367cd7f94f6ff0de8a6d02c723671263a5e49e361dd502b3dbc737073" => :yosemite
   end
 
+  devel do
+    url "https://www.zsh.org/pub/development/zsh-5.3.1-test-2.tar.gz"
+    version "5.3.1-test-2"
```
Comment (ilovezfs) [inline Formula/zsh.rb]: Not the best choice for a devel version string:
```
irb(main):001:0> Version.new("5.3.1-test-2") > Version.new("5.3.1")
=> true
```

Comment (zmwangx) [inline Formula/zsh.rb]: Well, yeah, but that's what the upstream uses.

Comment (ilovezfs) [inline Formula/zsh.rb]: Makes no difference. We need to override it with something that will be strictly less than stable.

Comment (zmwangx) [inline Formula/zsh.rb]: Wait a second. `5.3.1-test-2` is a release candidate for `5.4`. `5.3.1` is the current stable. Isn't this expected

```rb
[5] brew(main)> Version.new("5.3.1-test-2") > Version.new("5.3.1") && Version.new("5.3.1-test-2") < Version.new("5.4")
=> true
```

Comment (fxcoudert) [inline Formula/zsh.rb]: Devel should be higher version than stable, indeed

Comment (ilovezfs) [inline Formula/zsh.rb]: If 5.3.1-test-2 is a release candidate for 5.4, then it's fine.

**Comment by ilovezfs** [comment]
:shell:

**Pull request merged**